### DISCUSSION TITLE
WIMBoot problem with images > 3 GB?

### DISCUSSION THREAD

**Discussion by mobrembski** [ipxe/ipxe#741] [Q&A] [answered]
Hi!

I'm trying to make WinPE Sergiey Strelec edition to be bootable via PXE.
Mostly it works, but problem is that SS WinPE has drivers and applications in separate folder on USB, which i need to integrate into WIM image. Otherwise, i won't have any network and applications.
But problem is when i do it, then WIM image gets size more than 3 GB, which seems to be problematic for WIMBoot?
Or at least mine gets crashed when it triest to boot so big image (loads from NFS server ok). Its not a RAM Size prblem, i have 32 GB in my PC.

Question is, how to handle such situation?
Is WIMBoot able to use multiple WIM images, so i could split it ?
Or any other option?
Maybe mounting under running windows somehow?

**Comment by NiKiZe** [comment]
Do you get any error messages?
Are you booting in efi mode or pcbios mode?

It is likely a memory issue, it is not as simple as more RAM less issues. Often quite the opposite since memory maps gets more complex.

You can find similar already existing issues/discussions.

**Comment by mcb30** [comment]
@mobrembski A 3GB image is unlikely to work reliably on all systems.  Since the bulk of the space is almost certainly taken up by applications, and applications are not required in order to boot, I would suggest that you aim to include the network drivers within the WIM image and then serve the applications from a network share.

**Comment by mobrembski** [comment]
Yes, i see.

I know that it is very big but Sergiey Strelec Live WinPE is a special
case, i couldn't do it in other way :/

But I actually have this problem when i use Virtualbox VM, when i tried it
on real machines it works flawlessly :)
I think it can be closed.

wt., 13 wrz 2022 o 14:45 Michael Brown ***@***.***>
napisał(a):

**Comment by Mifody-07** [comment, marked as answer]
Hi, I'm uploading over the network, only the wim image takes up about 450MB and it doesn't put much load on RAM or my local network, to use applications I connect a network folder using a script - network.cmd 

my ipxe script for winpe strelec
:strelec
#set path
set path_file ${boot-url}/Files/Windows/strelec/
set path_ingect ${boot-url}/Files/Windows/ingect/
#set boot
kernel                           ${path_ingect}wimboot
imgfetch --name bcd              ${path_ingect}bcd                                     bcd
imgfetch --name boot.sdi         ${path_ingect}boot.sdi                                boot.sdi
iseq ${platform} pcbios && imgfetch --name bootmgr.exe ${path_ingect}pcbios/bootmgr.exe bootmgr.exe || imgfetch --name bootmgfw.efi ${path_ingect}efi/bootmgfw.efi bootmgfw.efi
imgfetch --name network.cmd      ${path_ingect}configs_strelec/strelec/network.cmd     network.cmd
imgfetch --name boot.wim         ${path_file}boot.wim                                  boot.wim
#end boot
boot || goto failed
goto start

network.cmd looks like that (I have a smb share in which winpe strelec applications are stored)
net use S: \\PXE\Windows\strelec /USER:PXE\SMB 12345678
